cfgov/unprocessed/apps/youth-employment-success/js/todo-notification.js

```diff
@@ -0,0 +1,124 @@
+const CLASSES = Object.freeze( {
+  CONTAINER: 'js-todo-notification',
+  MESSAGE: 'js-alert-content'
+} );
+
+const VISIBLE_CLASS = 'm-notification__visible';
+const SHOW_DEFAULT_MESSAGE = 'Item added to your to-do list.';
+const HIDE_DEFAULT_MESSAGE = 'Item removed from your to-do list.';
+
+/**
+ * TodoNotification
+ * @class
+ *
+ * @classdesc Handles behaviors around showing and hiding the
+ * 'added to to-do list' notification HTML.
+ *
+ */
+function TodoNotification() {
+  const self = this;
+
+  let cancel;
+  let clone;
+  let initialized = false;
+  let registeredNode;
+  let speed;
+
+  /**
+   * Remove notifications and clear timer
+   */
+  function _clearAll() {
+    clearTimeout( cancel );
+    _clearAlert();
+  }
+
+  /**
+   * Removes the notification node from the DOM
+   */
+  function _clearAlert() {
+    if ( clone ) {
+      registeredNode.removeChild( clone );
+      clone = null;
+      cancel = null;
+    }
+  }
+
+  /**
+   * Helper function to clone + append notification node
+   * @param {String} message The message the notification should display
+   */
+  function _cloneNotification( message ) {
+    clone = self.element.cloneNode( true );
+    clone.querySelector( `.${ CLASSES.MESSAGE }` ).textContent = message;
+    clone.classList.add( VISIBLE_CLASS );
+
+    registeredNode.appendChild( clone );
+  }
+
+  /**
+   * Show a notification.
+   * @param {String} message The optional message supplied to the notification
+   */
+  this.show = function show( message = SHOW_DEFAULT_MESSAGE ) {
```

Comment:
Since you have the underscore prefixed functions, I'd leave these as `function show( …`, and then below or above `this.element = null;` at the bottom have `this.show = show`, so that the public API is all aggregated together at the bottom of the class function.

Reply:
Ah gotcha, I will make those changes.